src/map/map_draw.cpp

```diff
@@ -256,6 +258,25 @@ void CViewport::DrawMapGridInViewport() const
 		Video.DrawLineClip(ColorDarkGray, {this->TopLeftPos.x - this->Offset.x, y}, {this->BottomRightPos.x, y});
 	}
 }
+/// Highlight overlays when size of logical and graphical tiles are identical
+void CViewport::HighlightGraphicalTileInViewport(int dx, int dy, uint32_t color /* = ColorRed */, uint8_t transparency /* = 64*/) const
+{
+	Video.FillTransRectangleClip(color, dx, dy, PixelTileSize.x, PixelTileSize.y, transparency);
+}
+
+/// Highlight overlays when size of logical and graphical tiles are differ
+void CViewport::HighlightGraphicalTileInViewport(int dx, int dy, int graphicTileOffset, int sx, 
```

Comment:
```suggestion
void CViewport::HighlightGraphicalTileInViewport(int dx, int dy, int graphicTileSize,
```
It seems to be **size** and not **offset**.

I think we can get rid of `sx` (which is for "optimization", as `sx = dx + dy * Map.Info.MapWidth`):
`Map.Fields[sx + j + (Map.Info.MapWidth * i)])` -> `Map.Fields(dx + i, dy + j)`

Reply:
> sx = dx + dy * Map.Info.MapWidth

Not really. `dx` and `dy` are in pixels, `sx` is the index of the top left tile which is at `(dx, dy)`.

Reply:
Anyway, I doubt the while thing about subtiles is working right:

We set a tile from the map file, with `SetTile(graphicTileIdx, tilePos, value, elevation)`
Say we have 4 logical tiles in a row per graphical tile (16 subtiles per whole graphical tile). 
1. And call `SetTile()` for tile with coordinates (0,0).
then here:
https://github.com/Wargus/stratagus/blob/a1bd75ec845eb36b6a55dcc2fe1a174a21f1b395/src/map/script_map.cpp#L805
we will call `*Map.Field(Vec2i(pos.x + j, pos.y + i))` for [0..3, 0..3].
2. Then we want to set next tile `SetTile()` for tile with coordinates (0,1) (call it from a map file). 
And we will call `*Map.Field(Vec2i(pos.x + j, pos.y + i))` for [0..3, 1..4]. So the the `y` range [1..3] will touched twice. And so on.

Reply:
or did I misunderstand the concept

Reply:
Yes, I just checked in stargus. `SetTile` x and y are in logical tile coordinates, but since you use it to set the graphical tile, you should skip however many logical tiles are within a graphical tile.

Reply:
> iirc (I may be misremembering) you just shouldn't call `SetTile(0, 0, ...)` and `SetTile(0, 1, ...)` if you have subtiles in e.g. stargus. You should call `SetTile(0, 0, ...)` and then `SetTile(0, 4, ...)` and so on.

Oh, thank you. 
Then there is a need to refine the editor - it doesn't support saving maps with subtiles right now. And check the map conversion code in stargus. 

And in general, I like the idea of subtiles for wargus too.